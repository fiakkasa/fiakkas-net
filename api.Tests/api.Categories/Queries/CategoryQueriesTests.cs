using api.Categories.Interfaces;
using api.Categories.Models;

namespace api.Categories.Queries;

public class CategoryQueriesTests
{
    [Fact]
    public void GetCategories_Should_Return_Data()
    {
        var item = new Category
        {
            Id = new Guid("38e483e4-6961-4b25-88a9-d1d0a5161109"),
            CreatedAt = new(2024, 1, 1, 0, 0, 0, TimeSpan.Zero),
            UpdatedAt = null,
            Version = 1,
            Title = "Title"
        };
        var dataRepository = new MockDataRepository<ICategory>([item]);
        var sut = new CategoryQueries();

        var result = sut.GetCategories(dataRepository);

        result.Should().NotBeEmpty();
        result.Should().BeAssignableTo<IQueryable<Category>>();
        result.First().Should().BeEquivalentTo(item);
        result.MatchSnapshot();
    }
}
