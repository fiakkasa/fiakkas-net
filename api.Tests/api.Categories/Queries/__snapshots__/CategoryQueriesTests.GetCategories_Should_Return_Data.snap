[
  {
    "Title": "Title",
    "Id": "c9f5879d-4018-49a0-9b71-b479dd5de7ff",
    "CreatedAt": "2024-01-01T00:00:00+00:00",
    "UpdatedAt": null,
    "Version": 1
  },
  {
    "Title": "Title",
    "Id": "38e483e4-6961-4b25-88a9-d1d0a5161109",
    "CreatedAt": "2024-01-01T00:00:00+00:00",
    "UpdatedAt": null,
    "Version": 1
  },
  {
    "Href": "/test",
    "Title": "Title",
    "Id": "ca832bf9-b7cb-4c31-bf8d-00f87a276fe3",
    "CreatedAt": "2024-01-01T00:00:00+00:00",
    "UpdatedAt": null,
    "Version": 1
  }
]
